src/main/java/virtual_pet/VirtualPet.java

```diff
@@ -0,0 +1,100 @@
+package virtual_pet;
+
+import java.util.Scanner;
+
+public class VirtualPet {
+    Scanner input = new Scanner(System.in);
+    String petName;
+    int hungerLevel = 6, happinessLevel = 6, sleepinessLevel = 6;
+    int angerLevel = 0; // value to track when a pet might leave if it gets neglected
+//  attribute levels are determined from 0-10 with 10 being fully satisfied
+//  and 0 being extreme need of attention.
+    public static VirtualPet createName(String name){
+        VirtualPet n = new VirtualPet();
+        n.petName = name;
+        return n;
+    }
```

Comment:
Method function createName should not exist.  You are recursively calling the VirtualPet object within itself and should not be done as this is a performance issue let alone not good software development practice.  This should be done in VirtualPetApplication and the name pass a parameter variable into the constructor which needs to exist.